applications/launchpad/backend/src/docker/mod.rs

```diff
@@ -57,3 +62,90 @@ lazy_static! {
 lazy_static! {
     pub static ref CONTAINERS: RwLock<HashMap<String, ContainerState>> = RwLock::new(HashMap::new());
 }
+
+pub fn tari_blockchain_volume_name(tari_workspace: &str, tari_network: TariNetwork) -> String {
+    format!("{}_{}_volume", tari_workspace, tari_network.lower_case())
+}
+
+pub fn network_name(tari_workspace: &str) -> String {
+    format!("{}_network", tari_workspace)
+}
+
+pub async fn try_create_container(
+    image: ImageType,
+    fully_qualified_image_name: String,
+    tari_workspace: &str,
+    config: LaunchpadConfig,
+    docker: Docker,
+) -> Result<ContainerCreateResponse, DockerWrapperError> {
+    debug!("{} has configuration object: {:#?}", fully_qualified_image_name, config);
+    let args = config.command(image);
+    let envars = config.environment(image);
+    let volumes = config.volumes(image);
+    let ports = config.ports(image);
+    let port_map = config.port_map(image);
+    let mounts = config.mounts(image, tari_blockchain_volume_name(tari_workspace, config.tari_network));
+    let mut endpoints = HashMap::new();
+    let endpoint = EndpointSettings {
+        aliases: Some(vec![image.container_name().to_string()]),
+        ..Default::default()
+    };
+    endpoints.insert(network_name(tari_workspace), endpoint);
+    let options = Some(CreateContainerOptions {
+        name: format!("{}_{}", tari_workspace, image.image_name()),
+    });
+    let config = Config::<String> {
+        image: Some(fully_qualified_image_name.to_string()),
+        attach_stdin: Some(false),
+        attach_stdout: Some(false),
+        attach_stderr: Some(false),
+        exposed_ports: Some(ports),
+        open_stdin: Some(true),
+        stdin_once: Some(false),
+        tty: Some(true),
+        env: Some(envars),
+        volumes: Some(volumes),
+        cmd: Some(args),
+        host_config: Some(HostConfig {
+            binds: Some(vec![]),
+            network_mode: Some("bridge".to_string()),
+            port_bindings: Some(port_map),
+            mounts: Some(mounts),
+            ..Default::default()
+        }),
+        networking_config: Some(NetworkingConfig {
+            endpoints_config: endpoints,
+        }),
+        ..Default::default()
+    };
+
+    Ok(docker.create_container(options, config).await?)
+}
+
+pub async fn try_destroy_container(image_name: &str, docker: Docker) -> Result<(), DockerWrapperError> {
+    let mut list_container_filters = HashMap::new();
+    list_container_filters.insert("name".to_string(), vec![image_name.to_string()]);
+    debug!("Searching for container {}", image_name);
+    let containers = &docker
+        .list_containers(Some(ListContainersOptions {
+            all: true,
+            filters: list_container_filters,
+            ..Default::default()
+        }))
+        .await?;
+
+    for c_id in containers.iter() {
+        let id = c_id.clone().id.unwrap();
```

Comment:
1. Is this `clone` necessary? Looks like you can avoid it
2. Is the unwrap infallible?